src/main/java/com/grimacegram/grimacegram/services/UserService.java

```diff
@@ -40,4 +42,18 @@ public Page<User> getUsers(User loggedInUser, Pageable pageable) {
         }
         return userRepository.findAll(pageable);
     }
+
+    public User getByUsername(String username) {
+        User inDB = userRepository.findByUsername(username);
+        if (inDB == null) {
+            throw new NotFoundException(username + "not found");
+        }
+        return inDB;
```

Comment:
FYI there is a library for mapping. Called ModelMapper. Really useful one. I advise you to consider using it for mapping from internal model classes to those you use to return data to the front-end.
For example you could map `User` to `UserVM` in a single line here.
That's not necessary just so that you know.